performance_search_max_throughput_test.py

```diff
@@ -0,0 +1,200 @@
+import re
+import json
+
+from collections import defaultdict
+from typing import Any
+
+# from sdcm.send_email import Email
+from performance_regression_test import PerformanceRegressionTest
+from sdcm.results_analyze import SearchBestThroughputConfigPerformanceAnalyzer
+from sdcm.utils.common import format_timestamp
+
+
+class MaximumPerformanceSearchTest(PerformanceRegressionTest):
+
+    def test_search_best_read_throughput(self):  # pylint: disable=too-many-locals
+        stress_params = {"stress_cmd_tmpl":  self.params.get('stress_cmd_r'),
+                         "test_name": "Test read"}
+        stress_params.update(**self._get_stress_parameters())
+
+        self.run_fstrim_on_all_db_nodes()
+        # run a write workload as a preparation
+        self.preload_data()
+        self.wait_no_compactions_running(n=400, sleep_time=120)
+        self.run_fstrim_on_all_db_nodes()
+
+        self.run_search_best_performance(**stress_params)
+
+    def test_search_best_write_throughput(self):  # pylint: disable=too-many-locals
+        stress_params = {"stress_cmd_tmpl":  self.params.get('stress_cmd_w'),
+                         "test_name": "Test write"}
+        stress_params.update(**self._get_stress_parameters())
+        # run a write workload as a preparation
+        self.preload_data()
```

Comment:
maybe stupid question, but if `preload_data` affects performance, shouldn't we recover cluster state after each stress loop? E.g. by recovering from backup etc.?

Reply:
Preload data runs only once. its goal is populate cluster with required dataset size.After that we work with this data set.
Because of data are not deleted (only updated and read) the size of data set is not changed.

Reply:
This is the main goal of test, find configuration with max throughput when scylla is under pressure of internal and external processes

Reply:
yes, but still, updating data increases read amplification and compaction load - I'm curious how much it affects results.